peoplehr/types.bal

```diff
@@ -0,0 +1,863 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+ 
+type Authentication record {
+    string APIKey?;
+    string Action?;
+};
+
+# Represents common fields of the response
+#
+# + isError - Indicates whether the response is an error or not
+# + Status - The status of the error  
+# + Message - The message of the response  
+public type PeopleHrResponse record {
+    boolean isError;
+    int Status;
+    string Message;
+};
+
+# Represents the response type of `getQueryByName` operation
+#
+# + Result - The results as an json array  
+public type GetQueryByNameResponse record {|
+    *PeopleHrResponse;
+    json Result;
+|};
+
+# Represents the response type of `getAllEmployeeDetail` operation
+#
+# + Result - All employee detail
+public type GetAllEmployeeDetailResponse record {|
+    *PeopleHrResponse;
+    json Result;
+|};
+
+# Represents the response type of `getEmployeeDetailById` operation
+#
+# + Result - The response details of the operation
+public type GetEmployeeDetailByIdResponse record {|
+    *PeopleHrResponse;
+    Employee? Result;
+|};
+
+# Represents the response type of `createNewEmployee` operation  
+public type CreateNewEmployeeResponse record {|
+    *UpdateEmployeeIdResponse;
+|};
+
+# Represents the response type of `updateEmployeeDetail` operation  
+public type UpdateEmployeeDetailResponse record {|
+    *UpdateEmployeeIdResponse;
+|};
+
+# Represents the response type of `updateEmployeeDetail` operation
+#
+# + Result - The response details of the operation 
+public type CheckAuthenticationResponse record {|
+    *PeopleHrResponse;
+    json Result;
+|};
+
+# Represents the response type of `UpdateEmployeeId` operation
+#
+# + Result - The response of the operation 
+public type UpdateEmployeeIdResponse record {|
```

Comment:
Shall we rename to `OperationStatus` and remove individual records which includes this alone?